cmd/soroban-rpc/internal/methods/health.go

```diff
@@ -42,10 +42,17 @@ func NewHealthCheck(retentionWindow uint32, ledgerRangeGetter LedgerRangeGetter,
 				Message: msg,
 			}
 		}
+		status := "healthy"
+		latestLedger := ledgerRange.LastLedger.Sequence
+		oldestLedger := ledgerRange.FirstLedger.Sequence
+		rententionWindowFull := (latestLedger - oldestLedger) >= retentionWindow
+		if rententionWindowFull {
+			status += ", retention window full"
```

Comment:
Are there a lot of places that do an `== "healthy"` check? This would break that. Maybe it'd be better to have a separate `RetentionWindowStatus: "full" | "incomplete"` kinda thing, instead?